### PR TITLE
Add ability to sort by several fields in requisitionsForConvert endpoint

### PR DESCRIPTION
**Pull request by lukaslew**
The _/requisitions/requisitionsForConvert_ endpoint accepts several **sortBy** parameters. Data returned by the endpoint will be sorted by those parameters in order of occurrence. By defaults data will be sorted by _emergency_ flag and _program name_.

---
This change is [Reviewable](https://reviewable.io/reviews/openlmis/openlmis-requisition/37)

### CODE REVIEW

File: src/main/resources/api-definition.yaml [hunk 1]
```diff
@@ -516,10 +516,10 @@ resourceTypes:
                   sortBy:
                       displayName: sortBy
                       description: This parameter contains field which we are going to sort by.
-                        There is three possibilities -  facilityCode, facilityName and programName
+                        There is four possibilities - emergency, facilityCode, facilityName and programName
```
Comment (pnaw94) [inline src/main/resources/api-definition.yaml]: Typo - there are four possibilities

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ Contract breaking changes:
 New functionality added in a backwards-compatible manner:
 
 * [OLMIS-2709](https://openlmis.atlassian.net/browse/OLMIS-2709): Changed ReferenceData facility service search endpoint to use smaller dto.
+* The /requisitions/requisitionsForConvert endpoint accepts several sortBy parameters. Data returned by the endpoint will be sorted by those parameters in order of occurrence. By defaults data will be sorted by emergency flag and program name.
```
Comment (pnaw94) [inline CHANGELOG.md]: This new default order should be also documented in api-definition, probably also javadoc

File: src/main/resources/api-definition.yaml (hunk 1 above)
Comment (lukaslew) [inline src/main/resources/api-definition.yaml]: Fixed

File: CHANGELOG.md (hunk 2 above)
Comment (lukaslew) [inline CHANGELOG.md]: I added information to api-definition and method javadoc

File: src/main/java/org/openlmis/requisition/service/RequisitionService.java [hunk 3]
```diff
@@ -508,7 +508,7 @@ public void checkIfCanApproveRequisition(UUID programId, UUID supervisoryNodeId,
   public Page<RequisitionWithSupplyingDepotsDto>
       searchApprovedRequisitionsWithSortAndFilterAndPaging(String filterValue,
                                                            String filterBy,
-                                                           String sortBy,
+                                                           List<String> sortBy,
```
Comment (mkwiatkowskisoldevelo) [inline src/main/java/org/openlmis/requisition/service/RequisitionService.java]: I think we should use sort parameters from Pageable class

Comment (lukaslew) [inline src/main/java/org/openlmis/requisition/service/RequisitionService.java]: Then I need to add sort field to **paginated** trait type (api-definition). The problem is that this type is used in 4 endpoints. I am not sure if all of them handle sorting.

Comment (mkwiatkowskisoldevelo) [inline src/main/java/org/openlmis/requisition/service/RequisitionService.java]: Can we add separate trait type 'sortalble'?

Comment (lukaslew) [inline src/main/java/org/openlmis/requisition/service/RequisitionService.java]: I created new trait type **sorted**and modified the code to use pageable class

File: src/main/java/org/openlmis/utils/BasicRequisitionDtoComparator.java [hunk 4]
```diff
@@ -25,74 +25,62 @@
...
-  /**
-   * Creates new instance with the passed compare conditions.
-   */
-  public BasicRequisitionDtoComparator(List<String> compareConditions) {
-    this.compareConditions = isEmpty(compareConditions)
-        ? Lists.newArrayList("emergency", "programName")
-        : compareConditions;
+    if (null == sort) {
+      sort = new Sort(
+          new Sort.Order(ASC, "emergency"), new Sort.Order(ASC, "programName")
```
Comment (pnaw94) [inline src/main/java/org/openlmis/utils/BasicRequisitionDtoComparator.java]: The comment says about "emergency" descending order being default, but it is different here.

Comment (lukaslew) [inline src/main/java/org/openlmis/utils/BasicRequisitionDtoComparator.java]: Updated

**Pull request merged**